### DISCUSSION TITLE
Set-PnPWeb for header background image

### DISCUSSION THREAD

**Discussion by dzfstephen** [pnp/powershell#1273] [Q&A] [answered]
Does Set_PnPWeb have property to change/update SharePoint Online Extended Header Background image?

**Comment by veronicageek** [comment, marked as answer]
@dzfstephen - It's actually hidden in the `*-PnPPropertyBag` cmdlet.
**However**, we've made it easier by creating a new command for it: `*-PnPWebHeader`. Update your module to the latest nightly release and try it out: `Update-Module pnp.powershell -AllowPrerelease -Force`

![image](https://user-images.githubusercontent.com/25181757/137978642-25da79fb-7fc1-4f84-966c-be756998b22a.png)

**Reply by dzfstephen** [reply]
@veronicageek  Thanks for your help. I tried to update the module. it mentioned V1.8.0' is in using, not V1.8.11. 
![image](https://user-images.githubusercontent.com/38846620/138034172-089e9178-946e-411b-902a-97368ce98458.png)
after I updated the version to V1.8.0, I still can not see the Get-PnPWebHeader or Set-PnPWebHeader. 
Any suggestions?

**Comment by AndersRask** [comment]
As Veroniqe mentioned, it is the latest build you need, so remember to add -AllowPrerelease to the cmdlet.

You might get an error since the modules is not catalog signed, so add -SkipPublisherCheck as well:

Install-Module -SkipPublisherCheck -AllowPrerelease -Force pnp.powershell

Regards
Anders Rask

Fra: dzfstephen ***@***.***>
Sendt: Wednesday, 20 October 2021 07.39
Til: pnp/powershell ***@***.***>
Cc: Subscribed ***@***.***>
Emne: Re: [pnp/powershell] Set-PnPWeb for header background image (Discussion #1273)


@veronicageek<https://github.com/veronicageek> Thanks for your help. I tried to update the module. it mentioned V1.8.0' is in using, not V1.8.11.
[image]<https://user-images.githubusercontent.com/38846620/138034172-089e9178-946e-411b-902a-97368ce98458.png>
after I updated the version to V1.8.0, I still can not see the Get-PnPWebHeader or Set-PnPWebHeader.
Any suggestions?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/pnp/powershell/discussions/1273#discussioncomment-1506135>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAMTDTRDHDK4VBX53C732LLUHZIVVANCNFSM5GIDFPTQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by dzfstephen** [reply]
Thanks Anders Rask.  I tried with the parameter -AllowPrerelease, and got some error. I will try it later. 
by the way, for the cmdlet Set-PnPWebHeader, what are the properties are? Can you please send me an example for updating the image cmdlet? 

Thanks & Regards
Stephen

**Comment by AndersRask** [comment]
It was right there in the reply I sent you:

You might get an error since the modules is not catalog signed, so add -SkipPublisherCheck as well:

Install-Module -SkipPublisherCheck -AllowPrerelease -Force pnp.powershell


Fra: dzfstephen ***@***.***>
Sendt: Wednesday, 20 October 2021 08.36
Til: pnp/powershell ***@***.***>
Cc: Anders Rask ***@***.***>; Comment ***@***.***>
Emne: Re: [pnp/powershell] Set-PnPWeb for header background image (Discussion #1273)


Thanks Anders Rask. I tried with the parameter -AllowPrerelease, and got some error. I will try it later.
by the way, for the cmdlet Set-PnPWebHeader, what are the properties are? Can you please send me an example for updating the image cmdlet?

Thanks & Regards
Stephen

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/pnp/powershell/discussions/1273#discussioncomment-1506322>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAMTDTU5WR6WGDEUKF5JV53UHZPMLANCNFSM5GIDFPTQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by dzfstephen** [reply]
thanks Anders Rask  and Veroniqe for your help. I installed the latest version on my machine. another question is how I can use the Get-PnPWebHeader and Set-PnPWebHeader. any documentation I can have a look?  I tried, but got the following error

![image](https://user-images.githubusercontent.com/38846620/138219041-0b207902-f6a9-45b3-a0d0-3e51b34a13d2.png)

Thanks 

Stephen

**Comment by veronicageek** [comment]
@dzfstephen - There's no parameters for the `Get-PnPWebHeader` but here's the docs for `Set-PnPWebHeader`: https://github.com/pnp/powershell/blob/dev/documentation/Set-PnPWebHeader.md
